accessible/tree.service.js

```diff
@@ -0,0 +1,325 @@
+/**
+ * AccessibleBlockly
+ *
+ * Copyright 2016 Google Inc.
+ * https://developers.google.com/blockly/
+ *
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Angular2 Service that handles all tree keyboard navigation.
+ * @author madeeha@google.com (Madeeha Ghori)
+ */
+blocklyApp.TreeService = ng.core
+  .Class({
+    constructor: function() {
+      console.log('making a new tree service');
+      this.activeDesc_ = {};
```

Comment:
Set this to Object.create(null), rather than {}. The former has no inherited properties that pollute your object, whereas the latter has 'toString' and a bunch of others that in theory could collide with your data.
Applies to anywhere an object is being used as a map.